### PR TITLE
chore(celery): Cleanup config and async query specifications

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

Whilst spelunking through the code looking at global async queries and Celery I realized that there was a number of inconsistencies with how the various `CeleryConfig` et al. objects were defined (referencing legacy settings or outdated tasks). This PR cleans up said logic.

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -1498,13 +1490,7 @@ def EMAIL_HEADER_MUTATOR(  # pylint: disable=invalid-name,unused-argument
 
 # Global async query config options.
 # Requires GLOBAL_ASYNC_QUERIES feature flag to be enabled.
-GLOBAL_ASYNC_QUERIES_REDIS_CONFIG = {
```
Comment (john-bodley) [inline superset/config.py:1501]: These were the same as the [defaults](https://github.com/redis/redis-py/blob/master/redis/client.py#L144-L187) and thus unncessary.

File: tests/integration_tests/superset_test_config.py [hunk 2]
```diff
@@ -133,11 +133,10 @@ def GET_FEATURE_FLAGS_FUNC(ff):
 
 
 class CeleryConfig:
-    BROKER_URL = f"redis://{REDIS_HOST}:{REDIS_PORT}/{REDIS_CELERY_DB}"
-    CELERY_IMPORTS = ("superset.sql_lab",)
-    CELERY_RESULT_BACKEND = f"redis://{REDIS_HOST}:{REDIS_PORT}/{REDIS_RESULTS_DB}"
-    CELERY_ANNOTATIONS = {"sql_lab.add": {"rate_limit": "10/s"}}
-    CONCURRENCY = 1
+    broker_url = f"redis://{REDIS_HOST}:{REDIS_PORT}/{REDIS_CELERY_DB}"
```
Comment (john-bodley) [inline tests/integration_tests/superset_test_config.py:136]: These config settings were [renamed](https://docs.celeryq.dev/en/stable/userguide/configuration.html#new-lowercase-settings) and will  be removed in Celery 6.0.

File: tests/integration_tests/superset_test_config_thumbnails.py [hunk 3]
```diff
@@ -62,10 +62,9 @@ def GET_FEATURE_FLAGS_FUNC(ff):
 
 
 class CeleryConfig:
-    BROKER_URL = f"redis://{REDIS_HOST}:{REDIS_PORT}/{REDIS_CELERY_DB}"
-    CELERY_IMPORTS = ("superset.sql_lab", "superset.tasks.thumbnails")
-    CELERY_ANNOTATIONS = {"sql_lab.add": {"rate_limit": "10/s"}}
-    CONCURRENCY = 1
+    broker_url = f"redis://{REDIS_HOST}:{REDIS_PORT}/{REDIS_CELERY_DB}"
```
Comment (john-bodley) [inline tests/integration_tests/superset_test_config_thumbnails.py:65]: These config settings were [renamed](https://docs.celeryq.dev/en/stable/userguide/configuration.html#new-lowercase-settings) and will be removed in Celery 6.0.

**Review by craig-rueda — APPROVED** [review]
LGTM

File: docs/docs/installation/async-queries-celery.mdx [hunk 4]
```diff
@@ -23,30 +23,17 @@ and web server processes should have the same configuration.
 
 ```python
 class CeleryConfig(object):
-    broker_url = 'redis://localhost:6379/0'
+    broker_url = "redis://localhost:6379/0"
     imports = (
-        'superset.sql_lab',
-        'superset.tasks',
+        "superset.sql_lab',
```
Comment (dpgaspar) [inline docs/docs/installation/async-queries-celery.mdx:28]: `"superset.sql_lab',` -> `"superset.sql_lab",`

**Review by dpgaspar — COMMENTED** [review]
just a small comment, thank for doing this!

**Review by dpgaspar — APPROVED** [review]

**Pull request merged**